### PR TITLE
MULE-12452: Fix Tests #2

### PR DESCRIPTION
**Pull request by juandesi**

### CODE REVIEW

File: core/src/main/java/org/mule/runtime/core/exception/ErrorTypeLocatorFactory.java [hunk 1]
```diff
@@ -71,7 +71,6 @@ public static ErrorTypeLocator createDefaultErrorTypeLocator(ErrorTypeRepository
             .addExceptionMapping(MessageRedeliveredException.class,
                                  errorTypeRepository.lookupErrorType(REDELIVERY_EXHAUSTED).get())
             .addExceptionMapping(Exception.class, unknown)
-            .addExceptionMapping(Throwable.class, unknown)
```
Comment (marianogonzalez) [inline core/src/main/java/org/mule/runtime/core/exception/ErrorTypeLocatorFactory.java]: why this?

File: tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/exception/HeisenbergConnectionExceptionEnricher.java [hunk 2]
```diff
@@ -15,6 +16,9 @@
 
   @Override
   public Exception enrichException(Exception e) {
+    if (e instanceof MuleFatalException) {
```
Comment (marianogonzalez) [inline tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/exception/HeisenbergConnectionExceptionEnricher.java]: why?

File: core/src/main/java/org/mule/runtime/core/exception/ErrorTypeLocatorFactory.java (hunk 1 above)
Comment (juandesi) [inline core/src/main/java/org/mule/runtime/core/exception/ErrorTypeLocatorFactory.java]: I reverted my previous change

File: tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/exception/HeisenbergConnectionExceptionEnricher.java (hunk 2 above)
Comment (juandesi) [inline tests/test-extensions/heisenberg-extension/src/main/java/org/mule/test/heisenberg/extension/exception/HeisenbergConnectionExceptionEnricher.java]: Why not? I dont want to enrich that kind of exception

**Pull request merged**